### PR TITLE
Additional MPAS-seaice plots for snow analyses

### PR DESCRIPTION
**Pull request by eclare108213**
Adding sea ice timeseries plots for snow volume and climatology plots for snow volume, broadband albedo, and ice area and volume tendencies due to thermodynamics and transport.

Checklist
* [x] User's Guide has been updated

* [x] Developer's Guide has been updated
* [x] API documentation in the Developer's Guide (`api.rst`) has any new or modified class, method and/or functions listed
* [x] Documentation has been [built locally](https://mpas-dev.github.io/MPAS-Analysis/latest/users_guide/quick_start.html#generating-documentation) and changes look as expected
* [x] `Testing` comment in the PR documents testing used to verify the changes

### CODE REVIEW

**Comment by eclare108213** [comment]
These plots are being added to enable a more comprehensive (but quick) analysis of an E3SM simulation campaign investigating the impacts of advanced snow physics processes in MPAS-seaice.  The fields being added here (snow volume, albedo and the area and volume tendencies) are fundamental when considering any sea ice simulations. 

For `Testing` please see
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.eclare/E3SMv3/MPAS_analysis/control/sea_ice/index.html
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.eclare/E3SMv3/MPAS_analysis/snow_grain/sea_ice/index.html
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.eclare/E3SMv3/MPAS_analysis/snow_redist/sea_ice/index.html
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.eclare/E3SMv3/MPAS_analysis/snow_off/sea_ice/index.html

Notes:

- In this first attempt to add new plots, I have not included observations of any kind.  This will need to come later, with input from the sea ice data experts.
- There is a lot of duplicative code in this PR.  For example, I would prefer to loop through all four tendencies in a single class, since they all need to be plotted exactly the same way.
- There are a few more fields that could be added, specific to snow.  Most of them are specific to only one simulation, with no data available in the other simulations.  I’m not sure how to handle this situation.
- I do not understand what sets the plot sizes on the web page.  Some of them are small, others are large.
- The color scales and other plot characteristics could be refined — this is a first attempt.
- I also cleaned up a few minor issues here and there.
- Once the code changes/additions in this PR are deemed suitable, I’ll work on the documentation.

How is authorship determined?  I am happy to include whomever should be acknowledged on those lines (understanding that I only modified Darin's existing code).

**Comment by xylar** [comment]
Sorry for the delay.  I'm very excited to review this!

**Comment by eclare108213** [comment]
I'm adding a few more fields now - hold off a bit please - but I would appreciate your suggestions for how it could be done better / more efficiently.

**Comment by eclare108213** [comment]
I've added ridging diagnostics and updated some of the other fields, and remade all the plots for 1950-1999.  E.g. see https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.eclare/E3SMv3/MPAS_analysis/1950-1999/snow_off/sea_ice/index.html

**Comment by xylar** [comment]
These additional plots look great!

One thing you may want to play around with is different colormaps for different fields.  I at least really like being able to quickly distinguish between different fields based on colormap.  There is a flag to `mpas_analysis` that makes it write out an image with all available colormaps:
```
mpas_analysis --plot_colormaps
```
That might be helpful.

**Comment by xylar** [comment]
> Unable to view documentation change made in mpas-dev on chrysalis

If you copy them somewhere on the web portal, they should be viewable.  If necessary, you can scp them to your laptop.  However, I'll take a look at the docs as part of my review so no worries.

**Comment by xylar** [comment]
> User's Guide has been updated

I think we need to add new entries in this directory for each new task.  You should be able to copy the documentation for an existing task and delete the part about observations (since my understanding is that the new tasks don't have observations yet).

https://github.com/MPAS-Dev/MPAS-Analysis/tree/develop/docs/users_guide/tasks

**Comment by eclare108213** [comment]
I updated the documentation on blues and scp'd it to my laptop to view.  I hope it's complete now!

**Comment by xylar** [comment]
Wonderful, thank you!  Looks good to me.

@darincomeau, do you have time to have a look at this in the next day or 2?

**Comment by xylar** [comment]
@darincomeau, if you can run analysis with this branch on one of the SORRM v3 simulations, that would be very helpful.  I'll then sign off on the PR.

**Comment by xylar** [comment]
@eclare108213, my only remaining comment would be that I think maybe this colormap might be fairly low contrast and thus it might be difficult to distinguish different ridged ice thicknesses from one another.  I grant you that the contouring makes the different levels pretty clear even so.  

image

I very much like your other colormaps you've chosen.

**Comment by xylar** [comment]
Thanks, @eclare108213, I think those levels are certainly easier to distinguish.  I imagine we will want to revisit the color maps and contour values down the road once we have some more experience with these plots across a variety of simulations.  That has certainly been the case with similar plots in the past.

File: docs/users_guide/tasks/climatologyMapSeaIceAlbedoSH.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
+.. _task_climatologyMapSeaIceAlbedoSH:
+
+climatologyMapSeaIceAlbedoSH
+============================
+
+An analysis task for plotting maps of Arctic sea ice broadband albedo.
```
Comment (xylar) [inline docs/users_guide/tasks/climatologyMapSeaIceAlbedoSH.rst:6]: A copy-and-paste error in the last commit:
```suggestion
An analysis task for plotting maps of Antarctic sea ice broadband albedo.
```

**Comment by darincomeau** [comment]
>@darincomeau, if you can run analysis with this branch on one of the SORRM v3 simulations, that would be very helpful. I'll then sign off on the PR.

Years 21 - 30 of the "alfred1" run:
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.dcomeau/E3SMv3_dev/20241121.v3.SORRME3r3.piControl.alfred1.chrysalis/mpas_analysis/eclare-snow/yrs21-30/sea_ice/index.html

**Review by darincomeau — APPROVED** [review]
@eclare108213 this is *alot* of work, and so great to have these additional plots!

**Review by xylar — APPROVED** [review]
Thanks @darincomeau, for your plots.

And, yes, thanks @eclare108213 for this tremendous work and for all you had to do to figure out MPAS-Analysis to make it happen.  This will make a huge difference in our ability to analyze sea ice going forward!

**Pull request merged**